core/src/main/kotlin/org/jetbrains/research/testspark/core/test/kotlin/KotlinTestCompiler.kt

```diff
@@ -0,0 +1,31 @@
+package org.jetbrains.research.testspark.core.test.kotlin
+
+import io.github.oshai.kotlinlogging.KotlinLogging
+import org.jetbrains.research.testspark.core.test.TestCompiler
+import org.jetbrains.research.testspark.core.utils.CommandLineRunner
+
+class KotlinTestCompiler(libPaths: List<String>, junitLibPaths: List<String>) :
+    TestCompiler(libPaths, junitLibPaths) {
+
+    private val log = KotlinLogging.logger { this::class.java }
+
+    override fun compileCode(path: String, projectBuildPath: String): Pair<Boolean, String> {
+        log.info { "[KotlinTestCompiler] Compiling ${path.substringAfterLast('/')}" }
+
+        val classPaths = "\"${getClassPaths(projectBuildPath)}\""
+        // Compile file
+        val errorMsg = CommandLineRunner.run(
+            arrayListOf(
+                "kotlinc",
+                "-cp",
+                classPaths,
+                path,
+            ),
+        )
+
+        log.info { "Error message: '$errorMsg'" }
+
+        // No need to save the .class file for kotlin, so checking the error message is enough
```

Comment:
When we run `kotlinc`, `.class` files will not be saved by default. We do not need to change the default behavior and save `.class` files, because `jacoco` will work without them (the situation is different for java)

Reply:
Wow, I did not expect a coverage utility to work without compiled sources. Okay, acknowledged that.